aptos-move/block-executor/src/captured_reads.rs

```diff
@@ -298,9 +325,14 @@ impl DelayedFieldRead {
 pub(crate) struct CapturedReads<T: Transaction> {
     data_reads: HashMap<T::Key, DataRead<T::Value>>,
     group_reads: HashMap<T::Key, GroupRead<T>>,
-    // Currently, we record paths for triggering module R/W fallback.
-    // TODO: implement a general functionality once the fallback is removed.
+    /// Captured module reads if V1 loader is used.
     pub(crate) module_reads: Vec<T::Key>,
+    /// Captured module reads if V2 loader is used. Here, using a vector is important for
```

Comment:
If it's not too troublesome, I'd much rather avoid this comment and have a hashmap that records the ModuleStorageReadResults directly and acts as a cache - i.e. in modules_and_scripts, we would first check if it's recorded and return, otherwise, we would fetch and record. We can project to ModuleStorageRead (or the new name) for validation purposes. This way CapturedReads acts as a single-threaded cache/consistency layer for the executing session, which is beneficial for reducing the surface of speculative inconsistencies, as well as generally good practice for performance (and consistent with what we do in captured reads for resources, etc)

Reply:
Good point, will do this

Reply:
Done